src/txmempool.h

```diff
@@ -549,6 +549,9 @@ class CTxMemPool
 
     std::vector<indexed_transaction_set::const_iterator> GetSortedDepthAndScore() const EXCLUSIVE_LOCKS_REQUIRED(cs);
 
+    /** track locally submitted transactions to periodically retry initial broadcast */
+    std::set<uint256> m_unbroadcast_txids GUARDED_BY(cs);
```

Comment:
I think this could be an unordered_set for constant-time insertion/lookup, but the size is going to be so small that it doesn't matter.

Reply:
ah, that makes sense. enforces the same uniqueness guarantees that I need, but the ordering is unnecessary.